File: 0050-powx-n/0050-powx-n.cpp
class Solution {
public:
    double pow(double x, int n){
        if(n == 0){
            return 1.0;
        }
        double half = pow(x,n/2);
        if(n % 2 == 0){
            
            return half*half;
        }
        else{
            return x*half*half;
        }
    }
    double myPow(double x, int n) {
        long N = n;
        if(N >= 0){
            return pow(x,N);
        }
        else{
            return  1.0/pow(x,-1l*N);
        }
    }
};

File: 0050-powx-n/0050-powx-n.java
class Solution {
    double pow(double x, long n){
        if(n == 0)return 1.0;
        if(n%2 == 1) 
        return pow(x, n-1)*x;
        double p= pow(x,n/2);
        return p*p;
    }
    public double myPow(double x, int n) {
        long ln = n;
        if(n >= 0)
        return pow(x,ln);
        
        return pow(1/x,-ln);

    }
}